fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/Select.java

```diff
@@ -35,33 +44,40 @@
     // The fields/expressions determining sort order. Optional
     private OrderByClause orderByClause;
 
+    // offset/limit for pagination
+    private PaginationClause paginationClause;
+
     /**
-     * Factory to create a new instance of the builder needed to create this
-     * statement
-     * 
-     * @return
+     * Default constructor. Not a DISTINCT select.
      */
-    public static SelectAdapter select(String... columns) {
-        return new SelectAdapter(columns);
+    protected Select() {
+    }
+
+    /**
+     * Protected constructor
+     * @param distinct if true, query will be SELECT DISTINCT ...
+     */
+    protected Select(boolean distinct) {
+        this.distinct = distinct;
     }
 
     /**
-     * Factory function for creating an {@link Alias} from a string
-     * 
-     * @param aliasStr
+     * Factory to create a new instance of the builder needed to create this
+     * statement
+     *
      * @return
      */
-    public static Alias alias(String aliasStr) {
-        return new Alias(aliasStr);
+    public static SelectAdapter select(String... columns) {
+        return new SelectAdapter(columns);
     }
 
-    public static Predicate predicate(String boolExpr) {
-        return new PredicateExpression(boolExpr);
+    public static SelectAdapter select(boolean distinct, String... columns) {
```

Comment:
general comment: not sure we're consistent about this, but should probably have method prologues for public methods

Reply:
we should...I added them. Was doing stuff too quickly when those were added.